### ISSUE TITLE
[postgres] Index USING → false negative

### ISSUE THREAD

**Issue by enote-kane** [sqlfluff/sqlfluff#3557]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Creating an index using a specific index method (which is not a function) gets mis-interpreted as a function name.

### Expected Behaviour

No error

### Observed Behaviour

```
sqlfluff lint --nocolor --verbose --dialect postgres 2022-07-05.index-using.sql
==== sqlfluff ====
sqlfluff:                1.1.0 python:                 3.8.10
implementation:        cpython verbosity:                   1
dialect:              postgres templater:               jinja
rules:                                all
==== readout ====

=== [ path: 2022-07-05.index-using.sql ] ===

== [2022-07-05.index-using.sql] FAIL                                                                                                                                              
L:   1 | P:  47 | L017 | Function name not immediately followed by parenthesis.
==== summary ====
violations:        1 status:         FAIL
All Finished!
```

### How to reproduce

Use the following file `2022-07-05.index-using.sql`:
```sql
CREATE INDEX my_json_idx ON my_table USING GIN (metrics_ jsonb_path_ops);
```

### Dialect

postgres

### Version

```
sqlfluff:                1.1.0 python:                 3.8.10
implementation:        cpython verbosity:                   1
dialect:              postgres templater:               jinja
rules:                                all
```

### Configuration

defaults

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
While it arguably might not technically be a function, is there any reason not to treat it as such for rule L017? Is it better to have no spaces here or not?

**Comment by enote-kane** [comment]
@tunetheweb Well, I think detecting the construct `GIN (...)` as a function here may lead to further bugs in more complex index definition cases.

So I think it would be a much more future-proof solution to detect and support the `USING ...` construct.

Adding a more advanced case shows that this is problematic:
```sql
CREATE INDEX my_json_idx ON my_table USING GIN (metrics_ jsonb_path_ops);
CREATE INDEX my_json_idx ON my_table USING BTREE (name_ COLLATE icu_std text_pattern_ops (op_variable = 'value') DESC NULLS LAST);
```

...result:
```
==== sqlfluff ====
sqlfluff:                1.1.0 python:                 3.8.10
implementation:        cpython verbosity:                   1
dialect:              postgres templater:               jinja
rules:                                all
==== readout ====

=== [ path: 2022-07-05.index-using.sql ] ===

== [2022-07-05.index-using.sql] FAIL                                                                                                                                              
L:   1 | P:  47 | L017 | Function name not immediately followed by parenthesis.
L:   2 | P:  49 | L017 | Function name not immediately followed by parenthesis.
L:   2 | P:  73 | L030 | Function names must be consistently upper case.
L:   2 | P:  89 | L017 | Function name not immediately followed by parenthesis.
L:   2 | P: 113 |  PRS | Line 2, Position 113: Found unparsable section: ' DESC
                       | NULLS LAST'
L:   2 | P: 131 | L016 | Line is too long.
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
==== summary ====
violations:        6 status:         FAIL
All Finished!
```

**Comment by tunetheweb** [comment]
OK and don't disagree there is more to do here for that unparsable section.

My question is more related to the original issue raised - in your opinion should L017 try to remove the space in `GIN (` or `BTREE (` regardless of whether it's called a function or not? Or best for L017 to ignore this space for these special `USING` "functions"?

Because if we stop this being recognised as a function, then either need to update L017 to re-include it again as whatever we call it (in which case, might be better leaving well enough alone), or we run the risk of someone else (that would prefer it to be treated as a function for L017) raising an issue saying it didn't remove the space.

I could go either way to be honest, so just trying to clarify the ask here.

**Comment by enote-kane** [comment]
In the long run, I'd personally prefer to implement proper handling of these constructs rather than as a function, but for the time being, I'd be OK with just ignoring the space between name and parenthesis for cases like these.
